pkg/hub/controllers/cluster/reconciler.go

```diff
@@ -165,24 +171,35 @@ func (r *Reconciler) updateClusterInfo(ctx context.Context, cr *hubv1alpha1.Clus
 		}
 	}
 
+	nodeIPs, err := cluster.GetNodeIP(r.MeshClient)
+	if err != nil {
+		log.Error(err, "Error Getting nodeIP")
+		return err
+	}
+
 	// Populate NodeIPs if not already updated
-	// Only needed to do the initial update. Later updates will be done by node reconciler
-	if cr.Spec.NodeIPs == nil || len(cr.Spec.NodeIPs) == 0 {
-		nodeIPs, err := cluster.GetNodeIP(r.MeshClient)
-		if err != nil {
-			log.Error(err, "Error Getting nodeIP")
-			return err
-		}
-		cr.Spec.NodeIPs = nodeIPs
-		if err := r.Update(ctx, cr); err != nil {
-			log.Error(err, "Error updating to cluster spec on hub cluster")
+	// OR gets the current nodeIP in use from controller cluster CR and compares it with
+	// the current nodeIpList (nodeIpList contains list of externalIPs or internalIPs)
+	// if the nodeIP is no longer available, we update the cluster CR on controller cluster
+	if cr.Status.NodeIPs == nil || len(cr.Status.NodeIPs) == 0 ||
+		!validatenodeipcount(nodeIPs, cr.Status.NodeIPs) {
+		log.Info("Mismatch in node IP", "IP in use", cr.Status.NodeIPs, "IP to be used", nodeIPs)
+		cr.Status.NodeIPs = nodeIPs
+		if err := r.Status().Update(ctx, cr); err != nil {
+			log.Error(err, "Error updating to cluster status on hub cluster")
 			return err
 		}
+		// TODO raise event to cluster CR that node ip is updated
 	}
-
 	return nil
 }
 
+// total -> external ip list of nodes in the k8s cluster
+// current -> ip list present in nodeIPs of cluster cr
+func validatenodeipcount(total, current []string) bool {
+	return reflect.DeepEqual(total, current)
```

Comment:
reflect.DeepEqual returns false even in case of order mismatch.. so instead we could check the exact elements present in both slices..